splitio/storage/pluggable.py

```diff
@@ -657,3 +678,167 @@ def clear(self):
+        self._tel_config = TelemetryConfig()
+        self._telemetry_config_key = 'SPLITIO.telemetry.init'
+        self._telemetry_latencies_key = 'SPLITIO.telemetry.latencies'
+        self._telemetry_exceptions_key = 'SPLITIO.telemetry.exceptions'
+        if prefix is not None:
+            self._telemetry_config_key = prefix + "." + self._telemetry_config_key
+            self._telemetry_latencies_key = prefix + "." + self._telemetry_latencies_key
+            self._telemetry_exceptions_key = prefix + "." + self._telemetry_exceptions_key
+
+    def _reset_config_tags(self):
+        """Reset config tags."""
+        with self._lock:
+            self._config_tags = []
+
+    def add_config_tag(self, tag):
+        """
+        Record tag string.
+
+        :param tag: tag to be added
+        :type tag: str
+        """
+        with self._lock:
+            if len(self._config_tags) < MAX_TAGS:
+                self._config_tags.append(tag)
+
+    def record_config(self, config, extra_config):
+        """
+        initilize telemetry objects
+
+        :param config: factory configuration parameters
+        :type config: Dict
+        :param extra_config: any extra configs
+        :type extra_config: Dict
+        """
+        self._tel_config.record_config(config, extra_config)
+
+    def pop_config_tags(self):
+        """Get and reset configs."""
+        with self._lock:
+            tags = self._config_tags
+            self._reset_config_tags()
+            return tags
+
+    def push_config_stats(self):
+        """push config stats to storage."""
+        self._pluggable_adapter.set(self._telemetry_config_key + "::" + self._sdk_metadata, str(self._format_config_stats()))
+
+    def _format_config_stats(self):
+        """format only selected config stats to json"""
+        config_stats = self._tel_config.get_stats()
+        return json.dumps({
+            'aF': config_stats['aF'],
+            'rF': config_stats['rF'],
+            'sT': config_stats['sT'],
+            'oM': config_stats['oM'],
+            't': self.pop_config_tags()
+        })
+
+    def record_active_and_redundant_factories(self, active_factory_count, redundant_factory_count):
+        """
+        Record active and redundant factories.
+
+        :param active_factory_count: active factory count
+        :type active_factory_count: int
+        :param redundant_factory_count: redundant factory count
+        :type redundant_factory_count: int
+        """
+        self._tel_config.record_active_and_redundant_factories(active_factory_count, redundant_factory_count)
+
+    def record_latency(self, method, latency):
+        """
+        record latency data
+
+        :param method: method name
+        :type method: string
+        :param latency: latency
+        :type latency: int64
+        """
+        self._method_latencies.add_latency(method, latency)
+        latencies = self._method_latencies.pop_all()['methodLatencies']
```

Comment:
why we need to pop things, you need to increment one bucket directly into the storage (same logic as you've added into exceptions)

Reply:
This is an intermediary step to get the correct formatted latency bucket, since the logic is already in Telemetry classes, you see that we add the method, then get the bucket immediately.

Reply:
The latencies have mappings of values, they are not direct values like exceptions

Reply:
why not doing this:
```
bucket = get_latency_bucket_index(latency)
latency_key = self._telemetry_latencies_key + '::' + self._sdk_metadata + '/' + method.value + '/' + str(bucket)
result = self._pluggable_adapter.increment(latency_key, 1)
```

Reply:
could you provide me an example of a `latency_key` created?

Reply:
Using inmemory dict for keys, example below:
{'SPLITIO.telemetry.latencies::python-9.4.0/1.1.1.1/ip1.1.1.1/0': 1}

your code looks good, will test it